### DISCUSSION TITLE
Chirped Pulse

### DISCUSSION THREAD

**Discussion by abigailalexjames** [epochpic/epoch#607] [Q&A]
Hi there, 

I am trying to add two chirped pulses. I found a discussion that suggested I just need to use the maths parser to vary the wavelength with time. This seems to work ok for the first pulse, however the one that follows appears to have an inverted field. Does this make sense? 

For reference here are my laser blocks for both:
`
begin:laser
boundary = x_min
lambda = lambda_probe + if(time lt 167*femto, lambda_probe*time/167.0e-15, 0.42*micron)
intensity_w_cm2 = I_probe*(w0_probe/wx_probe)^2
pol_angle = probe_polangle
profile = gauss(y, 0.0, wx_probe)
t_profile = 1
t_start = 0 * femto
t_end = 167 * femto
end:laser

begin:laser
boundary = x_min
lambda = lambda_probe + if(time gt 233*femto and time lt 400*femto, lambda_probe*(time-233.0e-15)/167.0e-15, 0.42*micron)
intensity_w_cm2 = I_probe*(w0_probe/wx_probe)^2
pol_angle = probe_polangle
profile = gauss(y, 0.0, wx_probe)
t_profile = 1
t_start = 233 * femto
t_end = 400 * femto
end:laser
`

And this is my output:
![chirp](https://github.com/Warwick-Plasma/epoch/assets/74933293/27aea07a-0083-4537-9c53-6d79663f2e8d)

I'm assuming that the shaping is correct as the pulses are gaussian, they propagate perpendicular to the boundary and are separated by 66 fs. 

Any input on this would be great. I expected to see two pulses that looked the same and not sure what has caused the difference. 
Thanks.

**Comment by DanRRRR** [comment]
How about adding to one of lasers one more line with some angle like pi
or whatever exactly the difference between them is because they start at different time:
```
phase = pi

**Reply by Status-Mirror** [reply]
I think Dan is right. From reading `laser.F90`, the magnitude of the laser fields at a given time is:

$E = A(t) \sin (\phi(t) + \int_0^t \omega(t')dt')$

Hence, it is not guaranteed that at two generic laser start-times $t_1$ and $t_2$ that the argument of the sine function would be the same. This explains why you have a difference.

As Dan says, you could tweak the `phase` parameter until the two curves have the same phase. It looks like `pi` would get you close, but to do this properly you would need to ensure that:

$\phi + \int_0^t \omega_2(t')dt' =  2\pi n + \int_0^t \omega_1(t')dt'$

where $\omega_1$ and $\omega_2$ are the frequency functions for the first and second laser respectively, and $n$ is some integer.

Hope this helps,
Stuart

**Reply by abigailalexjames** [reply]
This is great thanks!

I'll update on the answer once the HPC is working again and I can run to check!

**Reply by abigailalexjames** [reply]
Hey Stuart,

On the topic of my weird pulse shape:

I had an error in the input where I was adding lamda instead of delta_lambda. I've fixed the error in the input deck but the output still isn't as expected. If I plot the statement I expect to see a change in wavelength/frequency that would follow a linear increase as below:

![image](https://github.com/Warwick-Plasma/epoch/assets/74933293/9c26c980-988c-4dbf-a19a-b51dede2fd4d)
![image](https://github.com/Warwick-Plasma/epoch/assets/74933293/9fb4a935-dbb1-4276-894b-0ea2fe184930)

However when I run the deck the output pulse looks like this: 
![wonky_chirp](https://github.com/Warwick-Plasma/epoch/assets/74933293/de5b9e1e-3e4c-4746-aef0-95b23d3b1d01)


The laser block input is: 
`
```
begin:laser
boundary = x_min
lambda = lambda_probe + if(time lt 167e-15, 40e-9*time/167.0e-15, 0.42e-6)
intensity_w_cm2 = I_probe*(w0_probe/wx_probe)^2
phase = 0
pol_angle = probe_polangle
profile = gauss(y, 0.0, wx_probe)
t_profile = 1
t_start = 0 * femto
t_end = 167 * femto
end:laser
```
`
I tried changing the intensity value, adding a time profile, adjusting the cutoff to zero, ending the runtime at 167fs. Something is continuing to change the wavelength as the beam enters the domain and I cannot figure out why. 

I tried running a simple version of the example laser deck with the suggested lambda profile for the chirp using: 

```
``
begin:laser
    boundary = x_min
    intensity_w_cm2 = 1.0e18
    lambda = 1.0e-6 + if(time lt 100e-15, 1.0e-6*time/100e-15, 2e-6)
    profile = gauss(y,0,w_y)
    t_profile = gauss(time,t_hw01m,w_t)
end:laser

```

And this was the output:

![epoch_chirp](https://github.com/Warwick-Plasma/epoch/assets/74933293/81bd1ecb-ee2e-440a-80d3-ec4a91b04f8c)


I'm struggling to see why I can't replicate this just for a different pulse duration and frequency range. Any ideas?
